api/src/main/java/com/codeforcommunity/rest/ApiRouter.java

```diff
@@ -564,6 +576,46 @@ private void handleGetEventUsers(RoutingContext ctx) {
     }
     response.end(userJson);
   }
+  
+  private void handleResetPasswordEmail (RoutingContext ctx) {
+    HttpServerResponse response = ctx.response();
+    HttpServerRequest request = ctx.request();
+    String email = request.params().get("email"); 
+    
+    if(!email.endsWith("@husky.neu.edu")) {
+      response.setStatusCode(400).setStatusMessage("Incorrect email. Must be of type @husky.neu.edu.").end();
+    }
+    
+    //if email is valid send email
+    if(processor.validateEmail(email)) {
+      String token = createJWT(email, "auth-token", TOKEN_DURATION * 24, 0, true);
+      resetTokenMap.put(email, token);
+      EmailUtil.sendEmail(email, token);
```

Comment:
Could we implement rate limiting so that users can't be maliciously spammed with emails?

(Alternatively if you decide to keep the hashmap data store of tokens, could we make it so an email can only have one active reset password attempt at a time?)